oceannavigator/frontend/src/components/SubsetPanel.jsx

```diff
@@ -106,15 +137,38 @@ function SubsetPanel(props) {
       "&script_type=subset";
   };
 
+  // const getSubsetVariables = () => {
+  //   setLoading(true);
+  //   GetVariablesPromise(props.dataset.id).then((variableResult) => {
+  //     setLoading(false);
+  //     setSubset_variables(variableResult.data);
+  //   });
+  //   setOutputStarttime(props.dataset.starttime);
+  //   setOutputEndtime(props.dataset.time);
+  // };
+
   const getSubsetVariables = () => {
-    setLoading(true);
-    GetVariablesPromise(props.dataset.id).then((variableResult) => {
-      setLoading(false);
+  setLoading(true);
+
+  Promise.all([
+    GetVariablesPromise(props.dataset.id),
+    GetDepthsPromise(props.dataset.id, props.dataset.variable)
```

Comment:
Good idea to combine the promises here but since `props.dataset.variable` is inherited from the area window then the promise will never return any values if the user initially plotted a 2D variable. (Try opening an area plot with "Critical Depth" to see what I mean)

A safer approach might be to query depths when a 3d variable is selected from the variable `ComboBox` component by adding using a handler function with the onUpdate callback. The query might be cached from the users previous interactions and wouldn't be too inefficient if so. From there you can either store the depths in a state variable as you're doing or just rely on the cached data.